### DISCUSSION TITLE
Is it possible for the dashboard module count to include a filter?

### DISCUSSION THREAD

**Discussion by adam-jones-net** [area17/twill#1037] [Q&A] [answered]
I'm using the count feature on the dashboard modules that can be configured within twill.php.  Ie...

```
'dashboard' => [
        'modules' => [
              'App\Models\Property' => [
                'name' => 'properties',
                'label' => 'Properties',
                **'count' => true,**
```

The problem is that this does a total record count.  Sometimes this is useful, but in some cases this is really confusing to my client as they have a huge number of unpublished records and so the figure they see here is really not the figure they want.  They just need to see published records counted.  Is this possible ?

**Comment by pboivin** [comment]
Hey @arkid, I believe you could add  `'countScope' => 'published'` to the `'App\Models\Property'` array.

**Reply by pboivin** [reply]
Sorry, scratch this one! It doesn't work. Doing some quick testing on my end :)

**Reply by adam-jones-net** [reply]
Thanks @pboi20 .   It doesn't seem to work.   I added as referenced and got:

`Argument 2 passed to App\Repositories\PropertyRepository::filter() must be of the type array, string given, called in \vendor\area17\twill\src\Repositories\ModuleRepository.php on line 107`

So I added as '[published]' and then got:

```
Illuminate \ Database \ QueryException (42S22)
SQLSTATE[42S22]: Column not found: 1054 Unknown column '0' in 'where clause' (SQL: select count(*) as aggregate from `properties` where `0` = published and `properties`.`deleted_at` is null)
```

Any ideas?

**Reply by pboivin** [reply]
Quick fix: the syntax is actually `'countScope' => ['published' => true],`

**Comment by pboivin** [comment, marked as answer]
Cleaning up my answer for the records — You can add the `countScope` key for specific modules in the dashboard configuration:

```
'dashboard' => [
        'modules' => [
              'App\Models\Property' => [
                'name' => 'properties',
                'label' => 'Properties',
                'count' => true,
                'countScope' => ['published' => true],

        ...
```

This would restrict the count to published properties only.

**Reply by adam-jones-net** [reply]
Amazing! thanks!

**Comment by adam-jones-net** [comment]
See answer above
